src/server/common/resultsets/game_history.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#ifdef HAVE_MYSQLPP
+
+#include "server/common/resultsets/game_history.hpp"
+#include "serialization/string_utils.hpp"
+
+void game_history::read(mariadb::result_set_ref rslt)
+{
+    while(rslt->next())
+    {
+        result r;
+        r.game_name = rslt->get_string("GAME_NAME");
+        r.reload = rslt->get_boolean("RELOAD");
+        r.game_start = rslt->get_date_time("START_TIME").str();
+        r.scenario_name = rslt->get_string("SCENARIO_NAME");
+        r.scenario_id = rslt->get_string("SCENARIO_ID");
+        r.era_name = rslt->get_string("ERA_NAME");
+        r.era_id = rslt->get_string("ERA_ID");
+        for(const auto& player_info : utils::split(rslt->get_string("PLAYERS")))
+        {
+            std::vector<std::string> info = utils::split(player_info, ':');
+            if(info.size() == 2)
+            {
+                r.players.emplace_back(player{ info[0], info[1] });
+            }
```

Comment:
Perhaps this should be an assert or there should be an error message if the size is not 2.

Reply:
Which would you prefer?

Reply:
I think so far it can only happen if someone edits the DB manually? Still I'd rather be prepared for it to happen and print an error.

Reply:
Added an error logging statement.